pil-analyzer/src/evaluator.rs

```diff
@@ -461,10 +599,70 @@ mod internal {
                 };
                 Err(EvalError::FailedAssertion(msg))?
             }
+            BuiltinFunction::Print => {
+                let msg = match arguments.pop().unwrap().as_ref() {
+                    Value::String(msg) => msg.clone(),
+                    // As long as we do not yet have types, we just format any argument.
+                    x => x.to_string(),
+                };
+                print!("{msg}");
+                Value::Array(Default::default())
+            }
+            BuiltinFunction::ToInt => {
+                let arg = arguments.pop().unwrap().as_ref().clone();
+                Value::Integer(arg.try_to_integer()?)
+            }
+            BuiltinFunction::ToFe => {
+                let arg = arguments.pop().unwrap().as_ref().clone();
+                Value::FieldElement(arg.try_to_field_element()?)
+            }
+            BuiltinFunction::Modulus => Value::Integer(T::modulus().to_arbitrary_integer().into()),
         })
     }
 }
 
+pub fn evaluate_binary_operation_field<'a, T: FieldElement, C>(
+    left: T,
+    op: BinaryOperator,
+    right: T,
+) -> Value<'a, T, C> {
+    match op {
+        BinaryOperator::Add => Value::FieldElement(left + right),
+        BinaryOperator::Sub => Value::FieldElement(left - right),
+        BinaryOperator::Mul => Value::FieldElement(left * right),
```

Comment:
We currently do not have division or inversion in the field. I guess that would be quite useful. Should we just add it as `/` or should we use something different?

Reply:
maybe something else to make it more explicit? in the field lib?

Reply:
Let's either add it as a builtin or implemented function in https://github.com/powdr-labs/powdr/pull/929 - it already has the inverse for arbitrary prime fields.

Reply:
ok